portfolio/src/main/java/com/google/sps/servlets/TranslateAPIServlet.java

```diff
@@ -0,0 +1,30 @@
+package com.google.sps.servlets;
+
+import com.google.cloud.translate.Translate;
+import com.google.cloud.translate.TranslateOptions;
+import com.google.cloud.translate.Translation;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/translate")
+public class TranslateAPIServlet extends HttpServlet{
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String originalText = request.getParameter("text");
+        String langCode = request.getParameter("langCode");
+        System.out.println("Page translated too: " + langCode);
+
+        //translate text
+        Translate translate = TranslateOptions.getDefaultInstance().getService();
+        Translation translation = translate.translate(originalText, Translate.TranslateOption.targetLanguage(langCode));
```

Comment:
It would be good to validate the input language to make sure it is a language you support.
I was able to change the associated language from the console. So something like the code below would be nice. It's always good to validate input. That way you can also send back errors to the client (you don't need to do that now).

public class TranslateAPIServelet ... {
 private static Set<String> SUPPORTED_LANGUAGES = new HashSet<>(Arrays.asList("en", "es", "fr"));

public void doPost(...) {
  if (validateInput(request)) {
    // translate
  }
  // dont translate


}

Checking for language strings should be ok for now. But you could also use the java locale library - https://docs.oracle.com/javase/8/docs/api/java/util/Locale.html

Reply:
Good idea. I remember when I misspelled a language code my entire page was replaced with html responses.
As for the local, I'd love to look into that once I finish up with the other library.

Reply:
HAving a separate validate method, makes the code more readable. But ok for now.